### ISSUE TITLE
TSL handshake error

### ISSUE THREAD

**Issue by wilsonjc** [tobiasschuerg/InfluxDB-Client-for-Arduino#62]
I'm trying to set up a Secure Connection to an InfluxDB 1 server with an ESP8266, but I keep getting 
a TSL handshake error.    I was able to get a raspberry pi and grafana to connect to this InfluxDB server so I don't suspect that there is anything wrong with the InfluxDB setup.

 on the server and the following output from the ESP6266....

11:11:39.421 -> ......................[D] Validating connection to https://192.168.0.100:8086/ping?verbose=true
11:11:45.998 -> [D] HTTP status code - -1
11:11:45.998 -> [E] Error - connection refused
11:11:45.998 -> 0[D] Validating connection to https://192.168.0.100:8086/ping?verbose=true
11:11:46.098 -> [D] HTTP status code - -1
11:11:46.098 -> [E] Error - connection refused
11:11:46.098 -> InfluxDB connection failed: connection refused
11:11:46.098 -> Writing: wifi_status,device=ESP8266,SSID=Acanac_9XEzV rssi=-56i
11:11:46.098 -> [D] Flushing buffer: is oversized true, is timeout false, is buffer full false
11:11:46.098 -> [D] Prepare batch: bufferPointer: 1, batchPointer: 0, ceiling 1
11:11:46.098 -> [D] Prepare batch size 1
11:11:46.098 -> [D] Writing batch, size 1
11:11:46.098 -> [D] Writing to https://192.168.0.100:8086/write?db=test&u=james&p=Password123
11:11:46.098 -> [D] Sending:
11:11:46.098 -> wifi_status,device=ESP8266,SSID=Acanac_9XEzV rssi=-56i
11:11:46.098 -> 
11:11:46.198 -> [D] HTTP status code - -1
11:11:46.198 -> [E] Error - connection refused
11:11:46.198 -> [D] Leaving data in buffer for retry
11:11:46.198 -> InfluxDB write failed: connection refused
11:11:46.198 -> Wait 10s
11:11:56.232 -> Writing: wifi_status,device=ESP8266,SSID=Acanac_9XEzV rssi=-58i
11:11:56.232 -> [D] Flushing buffer: is oversized true, is timeout false, is buffer full false
11:11:56.232 -> [D] Prepare batch: bufferPointer: 2, batchPointer: 0, ceiling 2
11:11:56.232 -> [D] Prepare batch size 1
11:11:56.232 -> [D] Writing batch, size 1
11:11:56.232 -> [D] Writing to https://192.168.0.100:8086/write?db=test&u=james&p=Password123
11:11:56.232 -> [D] Sending:
11:11:56.232 -> wifi_status,device=ESP8266,SSID=Acanac_9XEzV rssi=-56i
11:11:56.232 -> 
11:11:56.332 -> [D] HTTP status code - -1
11:11:56.332 -> [E] Error - connection refused
11:11:56.332 -> [D] Leaving data in buffer for retry
11:11:56.332 -> InfluxDB write failed: connection refused
11:11:56.332 -> Wait 10s

****************************************************
here is the code..

[My_SecureWrite.zip](https://github.com/tobiasschuerg/InfluxDB-Client-for-Arduino/files/4684493/My_SecureWrite.zip)

**Comment by vlastahajek** [comment]
@wilsonjc, for connecting to the secured InfluxDB server is better to start with the [SecureWrite example](https://github.com/tobiasschuerg/InfluxDB-Client-for-Arduino/blob/master/examples/SecureWrite/SecureWrite.ino).
Your code is missing time synchronization, which is a critical condition for validating server certificate. 

On the other hand, [you can use SHA1 fingerprint](https://github.com/tobiasschuerg/InfluxDB-Client-for-Arduino#secure-connection) of the server (not CA) certificate, instead of a certificate. Verifying certificate fingerprint does not require time synchronization.

**Comment by wilsonjc** [comment]
I added the following code to the setup and the time is correct for my timezone but I am still getting the same errors....



 //configTime("CET-1CEST" , "pool.ntp.org", "time.nis.gov");
  configTime("EST5EDT" , "pool.ntp.org", "time.nis.gov");
  // Wait till time is synced
  Serial.print("Syncing time");
  int i = 0;
  while (time(nullptr) < 1000000000ul && i<100) {
    Serial.print(".");
    delay(100);
    i++;
  }
  Serial.println();

  time_t tnow = time(nullptr);
  Serial.println(ctime(&tnow));

**Comment by wilsonjc** [comment]
I finally used the SHA1 fingerprint (thank you vlastahajek) and it works perfectly.
In Ubuntu I used the following command to generate the kingerprint ...
 **sudo openssl x509 -noout -fingerprint -sha1 -inform pem -in  influxdb-selfsigned.crt**
and substituted the fingerprint that was generated into where I had the certificate.

const char InfluxDbCloud2CAFingerprint[] PROGMEM = "9B:62:0A:63:8B:B1:D2:CA:5E:DF:42:6E:A3:EE:1F:19:36:48:71:1F";


Thanks for all the help.
